Common/Data/Market/QuoteBar.cs

```diff
@@ -0,0 +1,420 @@
+                quoteBar.High = config.GetNormalizedPrice(csv[2].ToDecimal() / _scaleFactor);
+                quoteBar.Low = config.GetNormalizedPrice(csv[3].ToDecimal() / _scaleFactor);
+                quoteBar.Close = config.GetNormalizedPrice(csv[4].ToDecimal() / _scaleFactor);
+            }
+
+            quoteBar.AvgBidSize = csv[5].ToInt64();
+            return quoteBar;
+        }
+
+        /// <summary>
+        /// Parses forex quote bar data into the specified quotebar type, useful for custom types with OHLC data deriving from QuoteBar
+        /// </summary>
+        /// <typeparam name="T">The requested output type, must derive from QuoteBar</typeparam>
+        /// <param name="config">Symbols, Resolution, DataType, </param>
+        /// <param name="line">Line from the data file requested</param>
+        /// <param name="date">The base data used to compute the time of the bar since the line specifies a milliseconds since midnight</param>
+        /// <returns></returns>
+        public static T ParseForex<T>(SubscriptionDataConfig config, string line, DateTime date)
+            where T : QuoteBar, new()
+        {
+            var quoteBar = new T
+            {
+                Symbol = config.Symbol,
+                Period = config.Increment
+            };
+
+            var csv = line.Split(',');
+            if (config.Resolution == Resolution.Daily || config.Resolution == Resolution.Hour)
+            {
+                // hourly and daily have different time format, and can use slow, robust c# parser.
+                quoteBar.Time = DateTime.ParseExact(csv[0], DateFormat.TwelveCharacter, CultureInfo.InvariantCulture);
+                quoteBar.Open = Convert.ToDecimal(csv[1], CultureInfo.InvariantCulture);
+                quoteBar.High = Convert.ToDecimal(csv[2], CultureInfo.InvariantCulture);
+                quoteBar.Low = Convert.ToDecimal(csv[3], CultureInfo.InvariantCulture);
+                quoteBar.Close = Convert.ToDecimal(csv[4], CultureInfo.InvariantCulture);
+            }
+            else
+            {
+                //Fast decimal conversion
+                quoteBar.Time = date.Date.AddMilliseconds(csv[0].ToInt32());
+                quoteBar.Open = csv[1].ToDecimal();
+                quoteBar.High = csv[2].ToDecimal();
+                quoteBar.Low = csv[3].ToDecimal();
+                quoteBar.Close = csv[4].ToDecimal();
+            }
+            return quoteBar;
+        }
+
+
+        /// <summary>
+        /// Update the quotebar - build the bar from this pricing information:
+        /// </summary>
+        /// <param name="lastTrade">This trade price</param>
+        /// <param name="bidPrice">Current bid price (not used) </param>
+        /// <param name="askPrice">Current asking price (not used) </param>
+        /// <param name="volume">Volume of this trade</param>
+        public override void Update(decimal lastTrade, decimal bidPrice, decimal askPrice, decimal volume)
```

Comment:
I think it may be best to use the other Update method, this would require changing the signature of the `BaseData.Update` method to accept the bid/ask sizes